test/functional/rpc_net.py

```diff
@@ -96,5 +100,29 @@ def _test_getpeerinfo(self):
         assert_equal(peer_info[0][0]['addrbind'], peer_info[1][0]['addr'])
         assert_equal(peer_info[1][0]['addrbind'], peer_info[0][0]['addr'])
 
+    def _test_getnodeaddresses(self):
+        self.nodes[0].add_p2p_connection(P2PInterface())
+        network_thread_start()
+        self.nodes[0].p2p.wait_for_verack()
+
+        # send some addresses to the node via the p2p message addr
+        imported_addrs = set(["123.123." + str(i//255) + "." + str(i%255) for i in range(1000)])
+        now = int(time.time())
+        msg = msg_addr()
+        for a in imported_addrs:
+            addr = CAddress()
+            addr.time = now
+            addr.services = NODE_NETWORK | NODE_WITNESS
+            addr.ip = a
+            addr.port = 8333
+            msg.addrs.append(addr)
+        self.nodes[0].p2p.send_and_ping(msg)
+
+        # obtain addresses via rpc call and check they are a subset
+        NODE_ADDRESSES_REQUEST_COUNT = 10
+        node_addresses = self.nodes[0].getnodeaddresses(NODE_ADDRESSES_REQUEST_COUNT)
+        assert len(node_addresses) == NODE_ADDRESSES_REQUEST_COUNT
+        assert set([addr["address"] for addr in node_addresses]).issubset(imported_addrs)
```

Comment:
nit: drop the set comparisons, loop over the return items and test all fields, eg:

```python
        for a in node_addresses:
            assert a["address"] in imported_addrs
            assert_equal(a["services"], NODE_NETWORK | NODE_WITNESS)
            assert_equal(a["time"], now)
            assert_equal(a["port"], 8333)
```